### PR TITLE
tags: use py/mod tags for modules instead of py/repr

### PR DESCRIPTION
**Pull request by davvid**
A continuation of the ideas from #513 which we can refine after that PR is merged.

### CODE REVIEW

File: tests/datetime_test.py
Comment (Theelx) [inline tests/datetime_test.py:244]: Is there a reason not to use the `self.assertBoolean` functions? I would think it'd be easier as it would allow us to easily modify behavior of the functions if needed for whatever reason someday, plus for consistency with the rest of the existing tests.

File: jsonpickle/unpickler.py
Comment (Theelx) [inline jsonpickle/unpickler.py:22]: I have a gut feeling that this is going to come back to bite us in that loadmodule isn't compatible with some use case, but I can't find evidence for this feeling so YOLO I guess.

**Review by Theelx — APPROVED** [review]

File: tests/datetime_test.py
Comment (davvid) [reply tests/datetime_test.py:244]: Yeah, there's some logic to this, though it will look inconsistent from the perspective of the individual diffs.

The older camelCased `assertBoolean` methods are from the old days when the `unittest` module was new and the names were chosen to match the exact names of the Java junit library, which exposed a similar API.

One of pytest's selling points is that it lets us use a consistent naming convention by by leveraging the language's existing `assert` statements. pytest augments `assert` statements by adding pretty printing and diffing, which makes the advantages of using the junit-inspired methods almost non-existent. It does drop the ability to extend these methods, but I hope we won't need to do that :crossed_fingers: .

So over time I've taken the approach of replacing older tests' assertBoolean methods with pytest-native equivalents as I run into them updating the test method for some other reason. I'll just do this incrementally as tests get updated, though at some point I might make a series of file-by-file commits to updates the test suite all at once and leverage more of pytest's niceties. Until then the diffs will look like they're inconsistent with their surroundings, but it's okay since it should be incrementally going towards the newer pytest style.

While I find that `asserts` are shorter, simpler, and easier to read/type, it's mostly aesthetic as both ultimately do the same thing. The python core team decided that they'll keep the junit names around forever so there's not any pressing deprecation or technical reason to change it on any timeframe.

File: jsonpickle/unpickler.py
Comment (davvid) [reply jsonpickle/unpickler.py:22]: Yeah, this is probably the part that is worth a major version bump just for extra safety.

The question I faced here was, "should jsonpickle be safe by default, or opt-in to safety?". The correct answer should really be "safe by default," but it meant changing semantics. It's hard to know if anyone was relying on it so the changelog will need to be updated to call this out more.

Unfortunately we won't find out if this affects anyone until we put it out there and wait 3-5 years for all the slow upgraders to come out of the woodwork. As long as we have an easy way (`safe=False`) for users to opt-in to the compatible behavior then we should end up in a better place as they'll be consciously choosing to use the unsafe mode.

File: tests/datetime_test.py
Comment (Theelx) [reply tests/datetime_test.py:244]: Ah, that makes sense, thanks! I forgot that we use pytest, and from the POV of pytest it makes total sense to use asserts and not the unittest functions.

File: jsonpickle/unpickler.py
Comment (Theelx) [reply jsonpickle/unpickler.py:22]: I think you're right, jsonpickle should absolutely be safe by default. I do think that before this is merged, the CHANGES.rst file should be updated to specify that `safe=False` is an alternative if the user so chooses. I'll go ahead and update that now, and you can feel free to edit it and then merge it if you want.

**Comment by davvid** [comment]
I squashed in the typofix you committed and also squashed in a further update to the changelog that calls out the `safe=...` change as a breaking change.

Since I think this might warrant a major version bump we can hold onto these changes here and defer merging them until after the next v3.x release. We should probably let #515 play out and once once that's merged we'll tag the next v3.x release. These changes can follow behind that merge as the starting point for v4.x.

**Comment by davvid** [comment]
@Theelx I almost forgot about this one ;-)

I just rebased this against the latest and I think this is more or less about ready to go if you're still cool with the caveats we discussed in this PR.

**Pull request merged**
